src/style/colors.ts

```diff
@@ -323,7 +337,31 @@ function colorStringValue(color: string | object) {
   return color?.toString();
 }
 
-function adjustSaturation(colors: string[], color: string) {
+function adjustAllSaturations(colors: string[], color: string, levels: number[]) {
+  const array: string[] = [];
+  _.forEach(colors, (c, index) => {
+    if (c === color) {
+      array[index] = color;
+    } else {
+      const hsl = Color(c).hsl();
+      const saturation = hsl.color[1];
+      const level = levels[index];
+      if (level) {
+        const saturationLevel = saturation + level;
+        const clampedLevel = _.clamp(saturationLevel, 0, 100);
+        const adjusted = addSaturation(c, clampedLevel);
```

Comment:
Maybe it's out of scope but I think `addSaturation` is actually `setSaturation`

Reply:
Agree